cvat/apps/engine/backup.py

```diff
@@ -0,0 +1,354 @@
+import io
+import os
+from enum import Enum
+from zipfile import ZipFile
+
+from rest_framework.parsers import JSONParser
+from rest_framework.renderers import JSONRenderer
+
+import cvat.apps.dataset_manager as dm
+from cvat.apps.engine import models
+from cvat.apps.engine.serializers import (AttributeSerializer, DataSerializer,
+                                          FrameMetaSerializer,
+                                          LabeledDataSerializer,
+                                          SegmentSerializer,
+                                          SimpleJobSerializer, TaskSerializer)
+from cvat.apps.engine.utils import av_scan_paths
+
+
+class TaskExporter():
+    class Version(Enum):
+        V1 = 'v1'
+
+    MANIFEST_FILENAME = 'manifest.json'
+    DATA_DIRNAME = 'data'
+    TASK_DIRNAME = 'task'
+
+    def __init__(self, pk, version=Version.V1):
+        self._db_task = models.Task.objects.prefetch_related('data__images').select_related('data__video').get(pk=pk)
+        self._db_task.segment_set.all().prefetch_related('job_set')
+        self._db_data = self._db_task.data
+        self._version = version
+
+    def _write_directory(self, source_dir, zip_object, target_dir, exclude=None):
+        exclude = [] if exclude is None else exclude
+        for root, dirs, files in os.walk(source_dir, topdown=True):
+            dirs[:] = [d for d in dirs if d not in exclude]
+            for filename in files:
+                file_path = os.path.join(root, filename)
+                arcname = os.path.normpath(os.path.join(target_dir, os.path.relpath(root, source_dir), filename))
+                zip_object.write(filename=file_path, arcname=arcname)
+
+    def _write_data(self, zip_object):
+        data_dir = self._db_data.get_data_dirname()
+        self._write_directory(source_dir=data_dir, zip_object=zip_object, target_dir=self.DATA_DIRNAME)
+
+    def _write_task(self, zip_object):
+        task_dir = self._db_task.get_task_dirname()
+        self._write_directory(source_dir=task_dir, zip_object=zip_object, target_dir=self.TASK_DIRNAME, exclude=('artifacts', 'export_cache', 'logs'))
```

Comment:
Maybe we need to copy only specific dirs instead of excluding few names?
Maybe we need to take specific names from the task (eg. `task.get_task_dirname()`)?